File: app/src/lib/response/world.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

export interface GetRemoteWorldResponse {
  readonly config: WorldConfig;
  readonly createdBy: PlayerId;
  readonly hasPassword: boolean;
  readonly activePlayers: number;
  readonly currentRound: RoundId;
}

File: crates/nil-payload-macros/src/world.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use proc_macro::TokenStream;
use quote::quote;
use syn::DeriveInput;

pub fn impl_from_world(ast: &DeriveInput) -> TokenStream {
  let name = &ast.ident;
  let stream = quote! {
    #[automatically_derived]
    impl<T: Into<::nil_core::world::config::WorldId>> From<T> for #name {
      fn from(world: T) -> Self {
        Self { world: world.into() }
      }
    }
  };

  stream.into()
}
